contrib/builder/rpm/generate.sh

```diff
@@ -96,5 +109,10 @@ for version in "${versions[@]}"; do
 
 	echo 'ENV AUTO_GOPATH 1' >> "$version/Dockerfile"
 
-	echo 'ENV DOCKER_BUILDTAGS selinux' >> "$version/Dockerfile"
+	DOCKER_BUILDTAGS="selinux"
+	if [[ "$distro" == "amazon" ]]; then
+		DOCKER_BUILDTAGS="$DOCKER_BUILDTAGS !journald"
```

Comment:
Have you considering using the Oracle Linux 6 (`FROM oraclelinux:6`) image instead? That matches the Amazon Linux AMI far more closely than CentOS 7.